### DISCUSSION TITLE
[BUG] Enterprise Keywords Set Via Set-PnPTaxonomyFieldValue Are Removed When Editing Other Properties

### DISCUSSION THREAD

**Discussion by ashmsport** [pnp/powershell#3392] [General]
### Reporting an Issue or Missing Feature
Enterprise Keywords Set Via Set-PnPTaxonomyFieldValue Are Removed When Editing Other Properties
We have a document library with millions of documents. The properties are set via a Document based content type, one of the fields is Enterprise Keywords (renamed to Tags). I'd set enterprise keywords using Set-PnPTaxonomyFieldValue as there were so many documents which worked fine. But when another field of a document is edited the enterprise keyword automatically gets removed. It's intermittent. This does not happen when we set enterprise keywords manually. 
Opened a premier support case with Microsoft and they suggested checking with the community as the issue is only happening on enterprise keywords that were set via PnP PowerShell:
"_This has been extensively triaged and discussed internally with our escalation Team after which they have come up with the conclusion that since the issue only occurs when the item properties are changed using PnP PowerShell, this should be looked into by PnP Community support. 
This issue is not reproducible when changing file properties through GUI which has been confirmed by the customer. Issue only extends to one site and is not reproducible on any other site on the tenant.
Since PnP isn't a product. Therefore, it is not supported by official support channels.
PnP is supported in similar ways as other open-source projects done by Microsoft with support from the community by the community._
"

### Expected behavior 
Enterprise keywords should stay intact when other properties are edited

### Actual behavior 
When we updated a document property which has enterprise keyword set via PnP PowerShell the keyword disappears
Here's a file with the enterprise keyword 'Audits':
![image](https://github.com/pnp/powershell/assets/33230123/32460997-d89f-4aed-8b13-94a1332c4ae7)

When the Title property was manually edited (added a 9 in the end) in the details pane the keyword auto removed itself:
![image](https://github.com/pnp/powershell/assets/33230123/3670a629-d03d-4934-927e-80455b840668)

The same issue was reported [here](https://learn.microsoft.com/en-us/answers/questions/304433/sharepoint-online-enterprise-keywords-removed-afte?orderby=newest) by someone with videos

### Steps to reproduce behavior
```
#Parameters
$Tags = Import-Csv -Path "D:\Scripts\Tag0.csv"
$SiteURL = "https://tenantname.sharepoint.com/sites/OC"
$ListName = "Test2"

#Connect to the Site
Connect-PnPOnline -Url $SiteURL -Interactive

ForEach ($Tag in $Tags) 
{
        $TermName = $Tag.Tag
        $TermGroupName = "System"
        $TermSetName = "Keywords"
        $TermPath = "System|Keywords|" + $TermName
        $FileRelativeURL = $Tag.FileRelativeURL

        $Item = Get-PnPFile -Url $FileRelativeURL -AsListItem

        $Term = Get-PnPTerm -TermSet $TermSetName -TermGroup $TermGroupName -Identity $TermName

        #Set Tag
        Set-PnPTaxonomyFieldValue -ListItem $Item -InternalFieldName 'TaxKeyword' -TermId $Term.Id
        Write-Host -f White "Tag $($Term.Name) added as tag to $($FileRelativeURL)"
}
```
Tag0.csv file's columns and data are in the following format:
Tag,FileRelativeURL
 account,/sites/sitename/Clients/1-XYZ001/Account/Agreement.pdf
 Group_Details,/sites/sitename/Clients/1-ABC002/Account/LegalDocs/Group_Details/2004_RATE.doc

### What is the version of the Cmdlet module you are running?
    Directory: C:\Program Files\WindowsPowerShell\Modules

ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Manifest   2.1.1                 PnP.PowerShell                      Desk      {Add-PnPAdaptiveScopeProperty, Add-PnPP…
Manifest   1.12.0                PnP.PowerShell                      Core,Desk {Add-PnPAdaptiveScopeProperty, Add-PnPP…
Manifest   1.11.0                PnP.PowerShell                      Core,Desk {Add-PnPAdaptiveScopeProperty, Add-PnPP…

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows

**Comment by ashmsport** [comment]
This is also happening when copying files in bulk using Copy-PnpFile command.

**Reply by ashmsport** [reply]
@gautamdsheth Can you do me a favor. In the original [issue](https://github.com/pnp/powershell/issues/3325) where I reported the bug, in the steps to reproduce the behavior section I missed removing my company's tenant name. I'd like to remove it but can't as the issue has been closed. Can you reopen the issue and change it as below please?
$SiteURL = "https://tenantname.sharepoint.com/sites/OC"

**Reply by ashmsport** [reply]
@gautamdsheth Did you get a chance to look into this?

**Reply by gautamdsheth** [reply]
hi @ashmsport , I did look today.
Unfortunately, it looks like GitHub doesn't allow editing the issue.
These are the only 2 options that I receive.

<img width="660" alt="image" src="https://github.com/pnp/powershell/assets/8832018/5fe25c88-93df-4afa-a406-22fe2414efc8">
